comp/core/workloadmeta/collectors/internal/ecsfargate/ecsfargate.go

```diff
@@ -64,10 +70,27 @@ func (c *collector) Start(_ context.Context, store workloadmeta.Component) error
 		return err
 	}
 
+	if c.v4TaskEnabled {
+		c.metaV4, err = ecsmeta.V4FromCurrentTask()
+		if err != nil {
+			return err
+		}
+	}
+
 	return nil
 }
 
 func (c *collector) Pull(ctx context.Context) error {
+	if c.v4TaskEnabled {
```

Comment:
maybe here a switch case can be more elegant. 

```
switch c.APIVersion {
case aPIVersion4:
//...
case apiVersion2:
//...

}

Reply:
https://github.com/DataDog/datadog-agent/blob/8ebf82c86edf849ab4c9671c428196dc70d9c271/comp/core/workloadmeta/collectors/internal/ecsfargate/ecsfargate.go#L67-L79